spec/support/ssl_helper.rb

```diff
@@ -12,16 +12,41 @@ def initialize
     def next
       @iter.next
     rescue StopIteration
-      @keystash << OpenSSL::PKey::RSA.new(2048)
+      @keystash << generate_key
       save
       @keystash.last
     end
 
+    def generate_key
+      case (rand * 4).to_i
+      when 0
+        OpenSSL::PKey::RSA.new(2048)
+      when 1
+        generate_ecdsa_key('prime256v1')
+      when 2
+        generate_ecdsa_key('secp384r1')
+      when 3
+        generate_ecdsa_key('secp521r1')
+      end
+    end
+
+    def generate_ecdsa_key(curve)
+      k = OpenSSL::PKey::EC.new(curve)
+      k.generate_key
+      Acme::Client::CertificateRequest::ECKeyPatch.new(k)
```

Comment:
Plain `OpenSSL::PKey::EC` keys are pretty much unusable because they don't define `#private?`. [Here's](https://gist.github.com/mastahyeti/3fbb84a9fc482a0fae11c8bacc9d4d3a) the test output if you try.